h/api/models/document.py

```diff
@@ -182,6 +186,153 @@ def __repr__(self):
+        DocumentURI.type == type,
+        DocumentURI.content_type == content_type).first()
+
+    if docuri is None:
+        docuri = DocumentURI(claimant=claimant,
+                             uri=uri,
+                             type=type,
+                             content_type=content_type,
+                             document=document,
+                             created=created,
+                             updated=updated)
+        db.add(docuri)
+    elif not docuri.document == document:
+        log.warn('Found DocumentURI with id %d does not match expected '
+                 'document with id %d', docuri.document_id, document.id)
+
+    docuri.updated = updated
+
+
+def create_or_update_document_meta(db,
+                                   claimant,
+                                   claimant_normalized,
+                                   type,
+                                   value,
+                                   document,
+                                   created,
+                                   updated):
+    """
+    Create or update a DocumentMeta with the given parameters.
+
+    If an equivalent DocumentMeta already exists in the database then its value
+    and updated time will be updated.
+
+    If no equivalent DocumentMeta exists in the database then a new one will be
+    created and added to the database.
+
+    To be considered "equivalent" an existing DocumentMeta must have the given
+    claimant and type, but its value, document and created and updated times
+    needn't match the given ones.
+
+    :param db: the database session
+    :type db: sqlalchemy.orm.session.Session
+
+    :param claimant: the value to use for the DocumentMeta's claimant attribute
+        if a new DocumentMeta is created
+    :type claimant: unicode
+
+    :param claimant_normalized: the value of the new or existing DocumentMeta's
+        claimant_normalized attribute
+    :type claimant_normalized: unicode
+
+    :param type: the value of the new or existing DocumentMeta's type attribute
+    :type type: unicode
+
+    :param value: the value to set the new or existing DocumentMeta's value
+        attribute to
+    :type value: unicode
+
+    :param document: the value to use for the DocumentMeta's document if a new
+        DocumentMeta is created
+    :type document: h.api.models.Document
+
+    :param created: the value to use for the DocumentMeta's created attribute
+        if a new DocumentMeta is created
+    :type created: datetime.datetime
+
+    :param updated: the value to set the new or existing DocumentMeta's updated
+        attribute to
+    :type updated: datetime.datetime
+
+    """
+    existing_dm = DocumentMeta.query.filter(
+        DocumentMeta.claimant_normalized == claimant_normalized,
+        DocumentMeta.type == type).one_or_none()
+
+    if existing_dm is None:
+        db.add(DocumentMeta(
+            _claimant=claimant,
+            _claimant_normalized=claimant_normalized,
```

Comment:
You can skip having the `claimant_normalized` being passed-in to this function, and use the `claimant` (non-underscore) setter which takes care of setting the normalized claimant. (Similar to what you already did for the `create_or_update_document_uri` function).